src/transformers/trainer_tf.py

```diff
@@ -265,44 +285,40 @@ def train(self) -> None:
         logger.info("  Num Epochs = %d", epochs)
         logger.info("  Total optimization steps = %d", self.train_steps)
 
-        for epoch in range(start_epoch, int(epochs + 1)):
-            for training_loss in self._training_steps(train_ds, optimizer):
-                step = iterations.numpy()
+        self.global_step = 0  # used by loggers
```

Comment:
The global init here can even be removed.

Reply:
It is used if you want to eval several pre-trained models without additional training.

I can also change the logging logic to check if this parameter is present at all.

Reply:
We already set it to 0 in the ``__init__`